packages/analytics/src/client.ts

```diff
@@ -8,7 +8,7 @@ export function AnalyticsClient<T extends AnalyticsClientInterface>(
   return {
     async track<K extends keyof Events>(
       event: K,
-      ...properties: undefined extends Events[K] ? [] : [param: Events[K]]
+      properties?: undefined extends Events[K] ? [] : [param: Events[K]]
```

Comment:
I find it a bit confusing what this is doing

Reply:
Thanks @tigranpetrossian 👍 maybe with your update it's a bit clearer but sometimes I favour clarity over brevity. 

I haven't seen `undefined extends ` before so something new for me to learn.

Reply:
@pete-watters 
Sure. `extends` at type level just means "assignable" in TS, `undefined extends X`  is basically asking "is `undefined` a subset of type X`.